include/jemalloc/internal/extent_structs.h

```diff
@@ -117,15 +118,18 @@ struct extent_s {
 		size_t			e_bsize;
 	};
 
-	/*
-	 * List linkage, used by a variety of lists:
-	 * - arena_bin_t's slabs_full
-	 * - extents_t's LRU
-	 * - stashed dirty extents
-	 * - arena's large allocations
-	 * - arena's extent structure freelist
-	 */
-	ql_elm(extent_t)	ql_link;
+	union {
+		/*
+		 * List linkage, used by a variety of lists:
+		 * - arena_bin_t's slabs_full
+		 * - extents_t's LRU
+		 * - stashed dirty extents
+		 * - arena's large allocations
+		 */
+		ql_elm(extent_t)	ql_link;
+		/* Red-black tree linkage, used by arena's extent_avail. */
+		rb_node(extent_t)	rb_link;
+	};
```

Comment:
Unfortunately such an assertion would not hold in general, e.g. for the PGI compiler where ```RB_COMPACT``` is disabled to avoid a compiler limitation.